joomla/joomla-cms
# [5.1] Update Jooa11y with latest Sa11y build (Attempt #2)

Continuation of #42768 based on @Fedik's review and feedback. Includes updated screenshots of changes.

_(Sorry for re-creating the PR. Had issues re-basing from 5.0 to 5.1. After I renamed the branch on my fork, I inadvertently closed my PR.)_

### Summary of Changes
This PR adds [Sa11y](https://github.com/ryersondmp/sa11y) as a dependency, ensuring seamless and automatic updates for future Joomla releases. 

- Sa11y is added as a dependency via `build/build-modules-js/settings.json`. 
- Addition of several new languages. 
     - New languages would be added by modifying `build/build-modules-js/settings.json`. 
     - Automatic language detection based on page language.
     - Note: Only the admin settings will utilize Joomla's PLG language system.
- New configuration/properties in admin settings to customize experience for content editors.

Related issue: https://github.com/joomla-projects/joomla-a11y-checker/issues/75

### Testing Instructions
- Ensure you have a fresh JS build via `npm install` 
- Navigate to "Accessibility Check" as you would normally. Accessibility Checker should appear in same spot.
- Add a language pack (like German for example), and ensure Accessibility Checker displays correct translation.

### Actual result BEFORE applying this Pull Request
https://joomla-projects.github.io/joomla-a11y-checker/pages/errors.html

### Expected result AFTER applying this Pull Request

#### Accessibility Check preview
![View of updated accessibility checker in Accessibility Check preview.](https://github.com/joomla/joomla-cms/assets/26335725/2e68f84b-249f-4eab-9895-090d0967d886)

#### Plugin settings with descriptions
Includes a couple of **Extra Props** for demonstration purposes. Extra props can accept boolean/numeric/string based on [Sa11y's documentation](https://sa11y.netlify.app/developers/props/). 

![View of accessibility checker plugin admin settings with descriptions.](https://github.com/joomla/joomla-cms/assets/26335725/907ed751-607e-47a1-be09-45faa187e454)

### Link to documentations
Please select:

- [x] Pull Request link for manual.joomla.org: _I'll do a PR later to update this page https://manual.joomla.org/docs/accessibility/testing_
